mlir/lib/Target/LLVMIR/ModuleImport.cpp

```diff
@@ -1771,11 +1773,17 @@ ModuleImport::convertFunctionType(llvm::CallBase *callInst) {
   if (failed(calleeType))
     return failure();
 
-  // Compare the types to avoid constructing illegal call/invoke operations.
-  if (failed(verifyFunctionTypeCompatibility(*callType, *calleeType))) {
+  // Compare the types, if they are not compatible, avoid illegal call/invoke
+  // operations by issuing an indirect call. Note that LLVM IR currently
+  // supports this usage.
+  if (failed(checkFunctionTypeCompatibility(*callType, *calleeType))) {
     Location loc = translateLoc(callInst->getDebugLoc());
-    return emitError(loc) << "incompatible call and callee types: " << *callType
```

Comment:
Was there no test that covered this? 👀

Reply:
There is, the CHECK lines do not specifically check for `error`, so we got them for free. Let me add the `warning` part to make it more obvious.